pkg/osutil/osutil_test.go

```diff
@@ -106,6 +112,11 @@ func TestAllVariablesSubstituted(t *testing.T) {
 		{"{{mIxEdCase}}", true},
 		{"{{lowercase}}", true},
 		{"{{snake_case}}", true},
+		{"{{VAR1}} and {{VAR2}}", true},
+		{"{{.Values.name}} and {{.Values.namespace}}", false},
```

Comment:
This tells me gotemplate variables arent covered in the `checkAllVariablesSubstituted` function.  Looks like we should be using the gotemplate options with `"missingkey=error"` to check if we handle all variables correctly: https://pkg.go.dev/text/template#Template.Option